app/core/src/main/java/com/fsck/k9/mailstore/K9BackendStorage.kt

```diff
@@ -49,7 +49,10 @@ class K9BackendStorage(
         folderServerIds.asSequence()
                 .filterNot { account.isSpecialFolder(it) }
                 .map { localStore.getFolder(it) }
-                .forEach { it.delete() }
+                .forEach {
+                    it.delete()
+                    it.delete(true)
```

Comment:
`it.delete(); it.delete(true);` just... doesn't look nice.

If `delete` removes only folder settings maybe it would be better to rename it to `deleteSettings` or something like that.

The second one - `delete(true)` - it seems from my casual skimming of the code that the `true` refers to `recurse` parameter that is completely unused in all 4 implementations. Maybe that was something to be added or removed? Who knows, but currently it doesn't look good.

Reply:
Agreed. However, changing this touches a lot of areas unrelated to this fix. So I'd rather have this happen in a separate pull request.